### PR TITLE
MIG-802: legacy MTC operator for OCP 3.7 to 4.5 (MTC 1.5.1)

### PR DESCRIPTION
**Pull request by apinnick**
https://issues.redhat.com/browse/MIG-802

Legacy operator for OCP earlier than 4.6.
Installation and upgrade assemblies updated.
Compatibility guidelines: "latest 1.5.x" changed to 1.5.1.

1.5.1 GA: August 31.

4.6+

Previews:
https://deploy-preview-35761--osdocs.netlify.app/openshift-enterprise/latest/migrating_from_ocp_3_to_4/installing-3-4.html
https://deploy-preview-35761--osdocs.netlify.app/openshift-enterprise/latest/migrating_from_ocp_3_to_4/installing-restricted-3-4.html
https://deploy-preview-35761--osdocs.netlify.app/openshift-enterprise/latest/migration_toolkit_for_containers/installing-mtc.html
https://deploy-preview-35761--osdocs.netlify.app/openshift-enterprise/latest/migration_toolkit_for_containers/installing-mtc-restricted.html

https://deploy-preview-35761--osdocs.netlify.app/openshift-enterprise/latest/migrating_from_ocp_3_to_4/upgrading-3-4.html
https://deploy-preview-35761--osdocs.netlify.app/openshift-enterprise/latest/migration_toolkit_for_containers/upgrading-mtc.html

QE approved. Ready for peer review.

### CODE REVIEW

**Comment by apinnick** [comment]
@XinRedhat Ready for review.

File: modules/migration-upgrading-mtc-on-ocp-3.adoc [hunk 1]
```diff
@@ -23,14 +23,13 @@ You can upgrade {mtc-full} ({mtc-short}) on {product-title} 3 with `podman`.
...
 ----
 
-. Download the latest `operator.yml` file:
+. Download the `operator.yml` file:
 +
 [source,terminal,subs="attributes+"]
 ----
 $ sudo podman cp $(sudo podman create \
-  registry.redhat.io/rhmtc/openshift-migration-rhel7-operator:v{mtc-version}):/operator.yml ./ <1>
+  registry.redhat.io/rhmtc/openshift-migration-legacy-rhel8-operator:v1.5.1):/operator.yml ./
```
Comment (XinRedhat) [inline modules/migration-upgrading-mtc-on-ocp-3.adoc:31]: Same as above comments. `operator-3.7.yml` is only for OCP v3.7 clusters, `operator.yml `is for OCP v3.9 to OCP v4.5 clusters

File: modules/migration-upgrading-mtc-on-ocp-3.adoc [hunk 2]
```diff
@@ -71,12 +70,12 @@ $ oc scale -n openshift-migration --replicas=1 deployment/migration-operator
 $ oc -o yaml -n openshift-migration get deployment/migration-operator | grep image: | awk -F ":" '{ print $NF }'
 ----
 
-. Download the latest `controller-3.yml` file:
+. Download the `controller-3.yml` file:
```
Comment (XinRedhat) [inline modules/migration-upgrading-mtc-on-ocp-3.adoc:73]: There is only `controller.yml` in the legacy operator

File: modules/migration-upgrading-mtc-on-ocp-3.adoc [hunk 3]
```diff
@@ -71,12 +70,12 @@ $ oc scale -n openshift-migration --replicas=1 deployment/migration-operator
...
 +
 [source,terminal,subs="attributes+"]
 ----
 $ sudo podman cp $(sudo podman create \
-  registry.redhat.io/rhmtc/openshift-migration-rhel7-operator:v{mtc-version}):/controller-3.yml ./
+  registry.redhat.io/rhmtc/openshift-migration-legacy-rhel8-operator:v1.5.1):/controller-3.yml ./
```
Comment (XinRedhat) [inline modules/migration-upgrading-mtc-on-ocp-3.adoc:78]: There is only controller.yml in the legacy operator

File: migration_toolkit_for_containers/installing-mtc.adoc [hunk 4]
```diff
@@ -6,18 +6,18 @@ include::modules/common-attributes.adoc[]
 
 toc::[]
 
-You can install the {mtc-full} ({mtc-short}) on {product-title} 4.
+You can install the {mtc-full} ({mtc-short}) on {product-title} 4 in a networked environment.
```
Comment (mikemckiernan) [inline migration_toolkit_for_containers/installing-mtc.adoc:9]: Can you write more about the networked environment?  Is there a connectivity requirement?

If this is a conventionally installed cluster that is not in a restricted network, then I believe conventionally installed clusters do not receive any further clarification.

**Review by mikemckiernan — COMMENTED** [review]
nitty stuff.  If you have information regarding a team preference for "version" rather than "release"--as in the "OpenShift Container Platform version 3.11"--can you share it with me?

My instinct is that the customer-facing term would be [release](https://console.redhat.com/openshift/releases).

If I can clarify anything, please let me know.

File: migrating_from_ocp_3_to_4/installing-3-4.adoc [hunk 5]
```diff
@@ -43,17 +38,21 @@ The following storage providers are supported:
...
-[discrete]
-=== Additional resources for configuring a replication repository
+[id="replication-repository-prerequisites_{context}"]
+=== Prerequisites
 
-* xref:../migrating_from_ocp_3_to_4/about-mtc-3-4.adoc#migration-mtc-workflow_about-mtc-3-4[{mtc-short} workflow]
-* xref:../migrating_from_ocp_3_to_4/about-mtc-3-4.adoc#migration-understanding-data-copy-methods_about-mtc-3-4[About data copy methods]
-* xref:../migrating_from_ocp_3_to_4/migrating-applications-3-4.adoc#migration-adding-replication-repository-to-cam_migrating-applications-3-4[Adding a replication repository to the {mtc-short} web console]
+* All clusters must have uninterrupted network access to the replication repository.
+* If you use a proxy server with an internally hosted replication repository, you must ensure that the proxy allows access to the replication repository.
```
[the remark this answers is not in the document]
Comment (apinnick) [reply migrating_from_ocp_3_to_4/installing-3-4.adoc:45]: No. While this particular case is not covered in IBM, the Chicago Manual of Style excludes adverbs ending with 'y' from the rule regarding hyphenation of compound adverbs.

File: migration_toolkit_for_containers/installing-mtc.adoc (hunk 4 above)
Comment (apinnick) [reply migration_toolkit_for_containers/installing-mtc.adoc:9]: I could remove that. Just thought it seemed a little too terse.

File: migration_toolkit_for_containers/installing-mtc-restricted.adoc [hunk 6]
```diff
@@ -8,19 +8,17 @@ toc::[]
...
 You can install the {mtc-full} ({mtc-short}) on {product-title} 4 in a restricted network environment.
 
-[IMPORTANT]
-====
-You must install the same {mtc-short} version on all clusters.
-====
+You first install the {mtc-full} Operator on the {product-title} {product-version} target cluster with the Operator Lifecycle Manager.
 
-By default, the {mtc-short} web console and the `Migration Controller` pod run on the target cluster.
+Then you manually install the legacy {mtc-full} Operator on the {product-title} 4 source cluster.
```
[the remark this answers is not in the document]
Comment (apinnick) [reply migration_toolkit_for_containers/installing-mtc-restricted.adoc:13]: It's more of a requirement than an overview. But I did not want to turn it into a procedure by numbering the steps. The user has to install on OCP4 before installing on OCP3. I'll consider rewriting as "After installing x, ..."

**Comment by apinnick** [comment]
> nitty stuff. If you have information regarding a team preference for "version" rather than "release"--as in the "OpenShift Container Platform version 3.11"--can you share it with me?
> 
> My instinct is that the customer-facing term would be [release](https://console.redhat.com/openshift/releases).
> 
> If I can clarify anything, please let me know.

@mikemckiernan 

Interesting question. 

A search in the modules for "{product-title} version" found 85 results in 52 files, while "{product-title} release" found 20 results in 16 files.

Probably a good issue to submit to the OCP docs glossary.

**Pull request merged**